### PR TITLE
add oauth 2.0

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/oauth2.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- [ticloud auth login](/tidb-cloud/ticloud-auth-login.md): Authenticate with TiDB Cloud
+- [ticloud auth logout](/tidb-cloud/ticloud-auth-logout.md): Log out of TiDB Cloud
+
+### Authorization Code grant
+
+It is the most common OAuth 2.0 grant type. You can use it for both web apps and native apps to get an access token after a user authorizes an app.
+
+## Use OAuth to access TiDB Cloud
+
+If your app needs access to TiDB Cloud by using OAuth, submit a request to [become a Cloud & Technology Partner](https://www.pingcap.com/partners/become-a-partner/). We will reach out to you.
```
Comment (hfxsd) [inline tidb-cloud/oauth2.md:35]:
```suggestion
If your app needs access to TiDB Cloud by using OAuth, submit a request to [become a Cloud & Technology Partner](https://www.pingcap.com/partners/become-a-partner/) (select **Cloud & Technology Partner** in **Partner Program**). We will reach out to you.
```

File: tidb-cloud/oauth2.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## OAuth grant types
+
+The OAuth framework specifies several grant types for different use cases. TiDB Cloud supports the most common OAuth grant types, device code and authorization code.
+
+### Device Code grant
+
+It is usually used by browserless or input-constrained devices in the device flow to exchange a previously obtained device code for an access token.
+
+TiDB Cloud CLI supports OAuth 2.0 as follows:
```
Comment (hfxsd) [inline tidb-cloud/oauth2.md:24]:
```suggestion
TiDB Cloud CLI supports OAuth 2.0 Device Code grant type:
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: tidb-cloud/oauth2.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+### Device Code grant type
+
+It is usually used by browserless or input-constrained devices in the device flow to exchange a previously obtained device code for an access token.
+
+### Authorization Code grant type
+
+It is the most common OAuth 2.0 grant type, which enables both web apps and native apps to get an access token after a user authorizes an app.
+
+## Use OAuth to access TiDB Cloud
+You can access TiDB Cloud CLI using the OAuth 2.0 Device Code grant type:
```
Comment (qiancai) [inline tidb-cloud/oauth2.md:31]:
```suggestion

You can access TiDB Cloud CLI using the OAuth 2.0 Device Code grant type:
```

**Pull request merged**